octodns_transip/__init__.py

```diff
@@ -31,16 +33,48 @@ class TransipNewZoneException(TransipException):
     pass
 
 
+class TransipRetrieveRecordsException(ProviderException):
+    pass
+
+
+class TransipRetrieveNameserverException(ProviderException):
+    pass
+
+
+class TransipSaveRecordsException(ProviderException):
+    pass
+
+
+class TransipSaveNameserverException(ProviderException):
+    pass
+
+
 class TransipProvider(BaseProvider):
     SUPPORTS_GEO = False
     SUPPORTS_DYNAMIC = False
+    SUPPORTS_ROOT_NS = True
     SUPPORTS = set(
-        ('A', 'AAAA', 'CNAME', 'MX', 'NS', 'SRV', 'SPF', 'TXT', 'SSHFP', 'CAA')
+        (
+            'A',
+            'AAAA',
+            'CNAME',
+            'MX',
+            'NS',
+            'SRV',
+            'SPF',
+            'TXT',
+            'SSHFP',
+            'CAA',
+            'TLSA',
+            'NAPTR',
+            'ALIAS',
+            'DS',
+        )
     )
-    # unsupported by OctoDNS: 'TLSA'
     MIN_TTL = 120
     TIMEOUT = 15
     ROOT_RECORD = '@'
+    ROOT_NS_TTL = 3600  # Bogus value
```

Comment:
Maybe better to comment something like "See root NS handling in _process_desired_zone" for more information. Just seeing "Bogus value" here wouldn't really make sense to people who don't already know that the TTLs aren't configurable.

Reply:
Updated the comments to better inform the usage of the variable